### PR TITLE
Force the body parameter of `Patch` operations to be required

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/1968

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by AlexanderSher — CHANGES_REQUESTED** [review]
Blocking this one until we have an archboard decision about required body in PATCH and PUT
cc @m-nash @annelo-msft @tg-msft @JeffreyRichter

**Review by AlexanderSher — APPROVED** [review]
Null-checks for PATCH body parameter has been approved.

**Comment by ArcturusZhang** [comment]
Please preview the changes of this PR here: https://github.com/Azure/azure-sdk-for-net/pull/27299/files

File: test/TestProjects/MgmtDiscriminator/MgmtDiscriminator.json [hunk 1]
```diff
@@ -579,6 +579,7 @@
                 "cacheDuration": {
                     "description": "The duration for which the content needs to be cached. Allowed format is [d.]hh:mm:ss",
                     "type": "string",
+                    "format": "duration",
```
Comment (fengzhou-msft) [inline test/TestProjects/MgmtDiscriminator/MgmtDiscriminator.json:582]:
```suggestion
                    "format": "duration",
                    "x-ms-format": "duration-constant"
```
Based on the description, it's using the constant format not the default period format.

**Pull request merged**